### PR TITLE
feat(RHTAPREL-762): support multiple floating tags

### PR DESCRIPTION
**Pull request by johnbieren**
This commit modifies the push-snapshot task to accept a list of floatingTags in the data.images key of the RPA instead of just a single floatingTag string.

### CODE REVIEW

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: `| length` will return just the number, so both `-r` and `-c` are redundant here. (`-r` only makes sense for strings and `-c` only makes a difference if the result is a json, not just a single value.)

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: `-c` is redundant, the result is a list of strings (one per line), so nothing to compress here.

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: I think this needs a bit of improvement to say that this is for each of the floatingTags, like:

```sh
# If `floatingTags` is non-empty, for each of the `floatingTags` we push each image to $floatingTag and $floatingTag-$timestamp.
```

Maybe?

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: How about moving this one back to the main script to only call it once per component? With just two floatingTags, you will have 4 skopeo inspect calls for the same source image where it could have been just 1.

So I'd suggest moving the skopeo inspect call to something like like 104 (in the `for component...` loop, before determining what tags we need) and then adding the resulting digest as another parameter of the `push_image` function.

The only downside I can see is that we'll have two instances of `skopeo inspect` in two different places, but that should be fine.

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: `-c` is redundant.

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: It's plural now - common tags.

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [inline tasks/push-snapshot/push-snapshot.yaml]: I wonder if we should also push `"${floatingTag}-source". It seems weird that for binary images we will push both with and without timestamp, but for source, we only push with timestamp. I think we should ask somebody for clarification here. @ralphbean ?

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: fixed

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: fixed

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: updated

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: Smart, updated

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: fixed

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: fixed

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [reply tasks/push-snapshot/push-snapshot.yaml]: Cool!

File: tasks/push-snapshot/push-snapshot.yaml
Comment (mmalina) [reply tasks/push-snapshot/push-snapshot.yaml]: I'd like to add that I don't think this needs to be resolved here and now. So feel free to proceed with merging if you like. This can always be changed later.

**Review by mmalina — DISMISSED** [review]

File: tasks/push-snapshot/push-snapshot.yaml
Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: Cool, if CI passes before Ralph answers, I'll just merge and open a new PR based on his response

Comment (ralphbean) [reply tasks/push-snapshot/push-snapshot.yaml]: Yeah, tbh, it's not a requirement from the end user (legal) to use "${floatingTag}-source" but, I like the symmetry of it.  +1 from me to add that, but no need to hold up this change without it.

Comment (johnbieren) [reply tasks/push-snapshot/push-snapshot.yaml]: Updated to also push `${floatingTag}-source`

Comment (mmalina) [reply tasks/push-snapshot/push-snapshot.yaml]: Awesome, thanks to both of you!

**Review by mmalina — APPROVED** [review]

**Pull request merged**
